src/shared_modules/content_manager/src/actionOrchestrator.hpp

```diff
@@ -142,6 +125,71 @@ class ActionOrchestrator final
         m_spBaseContext->downloadedFileHash.clear();
     }
 
+    /**
+     * @brief Creates and triggers a new orchestration that updates the offset in the database.
+     *
+     * @param spUpdaterContext Updater context.
+     * @param offset New value of the offset.
+     */
+    void runOffsetUpdate(std::shared_ptr<UpdaterContext> spUpdaterContext, int offset) const
+    {
+        logDebug2(WM_CONTENTUPDATER, "Running '%s' offset update", m_spBaseContext->topicName.c_str());
+
+        if (0 > offset)
+        {
+            throw std::invalid_argument {"Invalid offset value: " + std::to_string(offset)};
+        }
+
+        spUpdaterContext->currentOffset = offset;
+
+        FactoryOffsetUpdater::create(m_spBaseContext->configData)->handleRequest(spUpdaterContext);
+    }
+
+    /**
+     * @brief Triggers a new orchestration that updates the content.
+     *
+     * @param spUpdaterContext Updater context.
+     * @param resetOffset If true, the current offset is set to zero.
+     */
+    void runContentUpdate(std::shared_ptr<UpdaterContext> spUpdaterContext, const bool resetOffset) const
+    {
+        logDebug2(WM_CONTENTUPDATER, "Running '%s' content update", m_spBaseContext->topicName.c_str());
+
+        // If the database exists, get the last offset
+        if (m_spBaseContext->spRocksDB)
+        {
+            spUpdaterContext->currentOffset = std::stoi(
+                m_spBaseContext->spRocksDB->getLastKeyValue(Components::Columns::CURRENT_OFFSET).second.ToString());
+        }
+
+        if (resetOffset)
+        {
+            spUpdaterContext->currentOffset = 0;
+        }
```

Comment:
Nitpick: Don't know how expensive the read operation over the DB is but if the resetOffset is present, then the operation is being performed without any need

```suggestion
        if (resetOffset)
        {
            spUpdaterContext->currentOffset = 0;
        }
        else
        {
            // If the database exists, get the last offset
            if (m_spBaseContext->spRocksDB)
            {
                spUpdaterContext->currentOffset = std::stoi(
                    m_spBaseContext->spRocksDB->getLastKeyValue(Components::Columns::CURRENT_OFFSET).second.ToString());
            }
        }
```

Also, I think we should have some fallback mechanism in case the DB doesn't exist (at least a log)

Reply:
> Nitpick: Don't know how expensive the read operation over the DB is but if the resetOffset is present, then the operation is being performed without any need

Good catch. Done.